### DISCUSSION TITLE
Why not have the REST API's "List commits" responses' commit schemas include the stats and array of files modified?

### DISCUSSION THREAD

**Discussion by ArchawinWongkittiruk** [community/community#30284] [Apps, API and Webhooks]
I just think it would be mighty convenient if the individual commit objects from the responses of "List commits" (https://docs.github.com/en/rest/commits/commits#list-commits) had the exact same schema as the responses from getting an individual commit i.e. "Get a commit" (https://docs.github.com/en/rest/commits/commits#get-a-commit), that is to say to similarly include the stats object and an array showing which files were modified!

I imagine this isn't the case at present in order to avoid 'bloating' the response schema of "List commits", which is a fair enough reason. However, a potentially decent workaround to this problem could be to make including the stats and file arrays optional additions to the schema by specifying such in the request query parameters, so something like '...?include_stats=true&include_files=true'.

What I'm doing at the moment instead to get the information I need is getting the commits for each filepath in the repo, which as you can imagine can get real slow and real expensive! If the "List commits" response schema included the modified files arrays, I would only have to make that one request to get all the information I need.

Please let me know what you think, and REST API developers if you're seeing this please consider making this a possibility! Thank you!

**Comment by leestott** [comment]
@martinwoodward would you be able to surface this with the right team? @ArchawinWongkittiruk  is a UCL MSc Students working on an industrial project with Microsoft.
